--- 
title: "Methods Comparison"
author: | 
  | Malu Calle Rosingana 
  | Coding: Yiwen Wang
date: "`r Sys.Date()`"
site: bookdown::bookdown_site
documentclass: book
bibliography: [book.bib]
biblio-style: apalike
link-citations: yes
description: "Methods comparison"
---

# Introduction 

This vignette provides a comparison of the 12 variables selected by the 3 methods:    

* logistic_lasso + linear constraints; 
* centered logratio transformation (CLR) + logistic_lasso; 
* selbal.

First, you will need to install then load the following packages:

## Packages installation and loading

```{r, message = FALSE, warning = FALSE}
#cran.packages <- c('knitr', 'MASS', 'VennDiagram', 'gplots', 'glmnet')
#install.packages(cran.packages)
#devtools::install_github(repo = "UVic-omics/selbal")

library(knitr) # rbookdown
library(MASS) # ginv
library(VennDiagram) # venn.diagram
library(gplots) # venn
library(glmnet) # glmnet
library(selbal) # selbal

# build in functions
source(file = 'functions.R')
```

## Simulations

The way we simulate datasets.

## Example datasets

We have two case studies.

### Microbiome and Crohn's disease

Crohn's disease (CD) is an inflammatory bowel disease that has been linked to microbial alterations in the gut [@gevers2014treatment; @oyri2015dysbiotic]. We use data from a large pediatric CD cohort study [@gevers2014treatment] to compare the proposed methodologies for identification of microbial signatures. 

Microbiome data from 16S rRNA gene sequencing and QIIME 1.7.0 bioinformatics processing were downloaded from Qiita [@rivera2018balances]. Only patients with Crohn's disease (n = 662) and those without any symptoms (n = 313) were analyzed. The OTU table was agglomerated to the genus level, resulting in a matrix with 48 genera and 975 samples.

We load the data that are provided in advance.

```{r}
load("./datasets/Crohn_data.rda")
# dim(x_Crohn)
# summary(y_Crohn)
```

|||
|:-------:|:-------:|
| No. of genera | 48 |
| No. of samples | 975 |
| No. of patients with CD | 662 |
| No. of healthy patients | 313 |
Table: **CD data summary**

### BEME day1 data

content need to be added.



