<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>Methods Comparison</title>
  <meta name="description" content="Methods comparison">
  <meta name="generator" content="bookdown  and GitBook 2.6.7">

  <meta property="og:title" content="Methods Comparison" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Methods comparison" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Methods Comparison" />
  
  <meta name="twitter:description" content="Methods comparison" />
  



<meta name="date" content="2019-05-14">

  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  

<link rel="next" href="coda.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Methods comparison</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#packages-installation-and-loading"><i class="fa fa-check"></i><b>1.1</b> Packages installation and loading</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#simulations"><i class="fa fa-check"></i><b>1.2</b> Simulations</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#example-datasets"><i class="fa fa-check"></i><b>1.3</b> Example datasets</a><ul>
<li class="chapter" data-level="1.3.1" data-path="index.html"><a href="index.html#microbiome-and-crohns-disease"><i class="fa fa-check"></i><b>1.3.1</b> Microbiome and Crohn’s disease</a></li>
<li class="chapter" data-level="1.3.2" data-path="index.html"><a href="index.html#beme-day1-data"><i class="fa fa-check"></i><b>1.3.2</b> BEME day1 data</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="coda.html"><a href="coda.html"><i class="fa fa-check"></i><b>2</b> CODA_LOGISTIC_LASSO</a><ul>
<li class="chapter" data-level="2.1" data-path="coda.html"><a href="coda.html#cd-data"><i class="fa fa-check"></i><b>2.1</b> CD data</a></li>
<li class="chapter" data-level="2.2" data-path="coda.html"><a href="coda.html#beme-day1-data-1"><i class="fa fa-check"></i><b>2.2</b> BEME day1 data</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="clr.html"><a href="clr.html"><i class="fa fa-check"></i><b>3</b> CLR_LOGISTIC_LASSO</a><ul>
<li class="chapter" data-level="3.1" data-path="clr.html"><a href="clr.html#cd-data-1"><i class="fa fa-check"></i><b>3.1</b> CD data</a></li>
<li class="chapter" data-level="3.2" data-path="clr.html"><a href="clr.html#beme-day1-data-2"><i class="fa fa-check"></i><b>3.2</b> BEME day1 data</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="selbal.html"><a href="selbal.html"><i class="fa fa-check"></i><b>4</b> Selbal: selection of balances</a><ul>
<li class="chapter" data-level="4.1" data-path="selbal.html"><a href="selbal.html#cd-data-2"><i class="fa fa-check"></i><b>4.1</b> CD data</a></li>
<li class="chapter" data-level="4.2" data-path="selbal.html"><a href="selbal.html#beme-day1-data-3"><i class="fa fa-check"></i><b>4.2</b> BEME day1 data</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="comparison.html"><a href="comparison.html"><i class="fa fa-check"></i><b>5</b> Concordance of variables selected by the three methods</a><ul>
<li class="chapter" data-level="5.1" data-path="comparison.html"><a href="comparison.html#cd-data-3"><i class="fa fa-check"></i><b>5.1</b> CD data</a></li>
<li class="chapter" data-level="5.2" data-path="comparison.html"><a href="comparison.html#beme-day1-data-4"><i class="fa fa-check"></i><b>5.2</b> BEME day1 data</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/EvaYiwenWang/Methods_comparison" target="blank">Source codes on github</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Methods Comparison</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="header">
<h1 class="title">Methods Comparison</h1>
<p class="author"><em><div style="white-space: pre-line;">Malu Calle Rosingana
Coding: Yiwen Wang</div></em></p>
<p class="date"><em>2019-05-14</em></p>
</div>
<div id="introduction" class="section level1">
<h1><span class="header-section-number">Chapter 1</span> Introduction</h1>
<p>This vignette provides a comparison of the 12 variables selected by the 3 methods:</p>
<ul>
<li>logistic_lasso + linear constraints;</li>
<li>centered logratio transformation (CLR) + logistic_lasso;</li>
<li>selbal.</li>
</ul>
<p>First, you will need to install then load the following packages:</p>
<div id="packages-installation-and-loading" class="section level2">
<h2><span class="header-section-number">1.1</span> Packages installation and loading</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#cran.packages &lt;- c(&#39;knitr&#39;, &#39;MASS&#39;, &#39;VennDiagram&#39;, &#39;gplots&#39;, &#39;glmnet&#39;)</span>
<span class="co">#install.packages(cran.packages)</span>
<span class="co">#devtools::install_github(repo = &quot;UVic-omics/selbal&quot;)</span>

<span class="kw">library</span>(knitr) <span class="co"># rbookdown</span>
<span class="kw">library</span>(MASS) <span class="co"># ginv</span>
<span class="kw">library</span>(VennDiagram) <span class="co"># venn.diagram</span>
<span class="kw">library</span>(gplots) <span class="co"># venn</span>
<span class="kw">library</span>(glmnet) <span class="co"># glmnet</span>
<span class="kw">library</span>(selbal) <span class="co"># selbal</span>

<span class="co"># build in functions</span>
<span class="kw">source</span>(<span class="dt">file =</span> <span class="st">&#39;functions.R&#39;</span>)</code></pre></div>
</div>
<div id="simulations" class="section level2">
<h2><span class="header-section-number">1.2</span> Simulations</h2>
<p>The way we simulate datasets.</p>
</div>
<div id="example-datasets" class="section level2">
<h2><span class="header-section-number">1.3</span> Example datasets</h2>
<p>We have two case studies.</p>
<div id="microbiome-and-crohns-disease" class="section level3">
<h3><span class="header-section-number">1.3.1</span> Microbiome and Crohn’s disease</h3>
<p>Crohn’s disease (CD) is an inflammatory bowel disease that has been linked to microbial alterations in the gut <span class="citation">(Gevers et al. <a href="#ref-gevers2014treatment">2014</a>; Øyri, Műzes, and Sipos <a href="#ref-oyri2015dysbiotic">2015</a>)</span>. We use data from a large pediatric CD cohort study <span class="citation">(Gevers et al. <a href="#ref-gevers2014treatment">2014</a>)</span> to compare the proposed methodologies for identification of microbial signatures.</p>
<p>Microbiome data from 16S rRNA gene sequencing and QIIME 1.7.0 bioinformatics processing were downloaded from Qiita <span class="citation">(Rivera-Pinto et al. <a href="#ref-rivera2018balances">2018</a>)</span>. Only patients with Crohn’s disease (n = 662) and those without any symptoms (n = 313) were analyzed. The OTU table was agglomerated to the genus level, resulting in a matrix with 48 genera and 975 samples.</p>
<p>We load the data that are provided in advance.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">load</span>(<span class="st">&quot;./datasets/Crohn_data.rda&quot;</span>)
<span class="co"># dim(x_Crohn)</span>
<span class="co"># summary(y_Crohn)</span></code></pre></div>
<table>
<caption><strong>CD data summary</strong></caption>
<tbody>
<tr class="odd">
<td align="center">No. of genera</td>
<td align="center">48</td>
</tr>
<tr class="even">
<td align="center">No. of samples</td>
<td align="center">975</td>
</tr>
<tr class="odd">
<td align="center">No. of patients with CD</td>
<td align="center">662</td>
</tr>
<tr class="even">
<td align="center">No. of healthy patients</td>
<td align="center">313</td>
</tr>
</tbody>
</table>
</div>
<div id="beme-day1-data" class="section level3">
<h3><span class="header-section-number">1.3.2</span> BEME day1 data</h3>
<p>content need to be added.</p>

</div>
</div>
</div>
<h3>References</h3>
<div id="refs" class="references">
<div id="ref-gevers2014treatment">
<p>Gevers, Dirk, Subra Kugathasan, Lee A Denson, Yoshiki Vázquez-Baeza, Will Van Treuren, Boyu Ren, Emma Schwager, et al. 2014. “The Treatment-Naive Microbiome in New-Onset Crohn’s Disease.” <em>Cell Host &amp; Microbe</em> 15 (3). Elsevier: 382–92.</p>
</div>
<div id="ref-oyri2015dysbiotic">
<p>Øyri, Styrk Furnes, Györgyi Műzes, and Ferenc Sipos. 2015. “Dysbiotic Gut Microbiome: A Key Element of Crohn’s Disease.” <em>Comparative Immunology, Microbiology and Infectious Diseases</em> 43. Elsevier: 36–49.</p>
</div>
<div id="ref-rivera2018balances">
<p>Rivera-Pinto, J, JJ Egozcue, Vera Pawlowsky-Glahn, Raul Paredes, Marc Noguera-Julian, and ML Calle. 2018. “Balances: A New Perspective for Microbiome Analysis.” <em>MSystems</em> 3 (4). Am Soc Microbiol: e00053–18.</p>
</div>
</div>
            </section>

          </div>
        </div>
      </div>

<a href="coda.html" class="navigation navigation-next navigation-unique" aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"download": ["Methods_comparison.pdf"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:" && /^https?:/.test(src))
      src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
